### PR TITLE
Document can execute web

### PR DESCRIPTION
**Pull request by Noah-Silvera**
Document the command canExecute pattern for Geocortex Web

* Add a section in the component reference and service reference for canExecute
* Add a tutorial with a command that uses the canExecute pattern

### CODE REVIEW

File: docs/web/tutorial-command-can-execute.mdx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            this.messages
+                .command("custom.command-with-can-execute")
+                .register(() => {
+                    this.messages.commands.ui.alert.execute({ message: "Executed `custom.command-with-can-execute`"});
+                    return;
+                },
+                // highlight-start
+                () => {
+                    return this._canExecuteValue;
+                }))
```
Comment (eric-parsons) [inline docs/web/tutorial-command-can-execute.mdx:292]:
```suggestion
                () => this._canExecuteValue
            ))
```

File: docs/web/snippets/can-execute.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+this.messages
+    .command("custom.command-with-can-execute")
+    .canExecuteChanged.subscribe(() => {
+        // .. react to can execute status changing
+    });
+```
+
+:::tip
+For a full example of a command with a dynamic execution status, [checkout the `canExecute` tutorial](../tutorial-command-can-execute.mdx).
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/web/snippets/can-execute.mdx]: Fixed here and anywhere else in the docs this occurred

File: docs/web/snippets/can-execute.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    // highlight-start
+    (x: number) => {
+        return x != 0;
+    }
+);
+// highlight-end
+```
+
+If the `canExecute` hook returns true, the command will be executed. If the function returns false, the command will not be executed.
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/web/snippets/can-execute.mdx]: Excellent points. Rephrased this paragraph to be more specific about how a canExecute callback appllies to a specific implemtentation of a command.

File: docs/web/snippets/can-execute.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+```tsx
+this.messages.command<number>("custom.divide-ten-by-x").register(
+    (x: number) => {
+        this.messages.commands.ui.alert.execute({
+            message: `10 / ${x} is ${10 / x}`,
+        });
+    },
+    // highlight-start
+    (x: number) => {
+        return x != 0;
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/web/snippets/can-execute.mdx]: nope... fixed.

File: docs/web/tutorial-command-can-execute.mdx [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    LayoutElement,
+    LayoutElementProperties,
+} from "@geocortex/web/components";
+import Button from "@geocortex/web/ui/button";
+import React from "react";
+
+import CustomModel from "./CustomModel";
+
+const CustomComponent = (props: LayoutElementProperties<CustomModel>) => {
+    const { model } = props;
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/web/tutorial-command-can-execute.mdx]: Removed.

**Comment by Noah-Silvera** [comment]
> FYI we have a related sample that could be useful. It demonstrates canExecute as well.
> 
> https://vertigis-web-samples.netlify.app/commands-and-operations

Thanks for pointing that out! Added some links to the sample where appropriate

**Pull request merged**